src/extension.ts

```diff
@@ -1067,10 +1069,26 @@ class ChatMessageBuilder {
 		// Ignore empty messages
 		if (message.trim().length === 0) { return; }
 
-		this.chatMessages.push({
-			role: role,
-			content: message,
-		});
+		// Matches images in Markdown (i.e., `![alt](url)`)
+		const parts = message.split(/(!\[[^\]]*\]\([^)]+\))/gm).filter(part => part !== '');
+
+		if (parts.length === 1) {
+			this.chatMessages.push({ role: role, content: message });
+		} else {
+			const content: OpenAI.ChatCompletionContentPart[] = parts.map<OpenAI.ChatCompletionContentPart>(part => {
+				const url = part.match(/!\[[^\]]*\]\(([^)]+)\)/)?.[1];
+				if (url === undefined) { return { type: 'text', text: part }; }
+				if (url.match(/^https?:\/\//)) {
+					return { type: 'image_url', image_url: { url: url } };
+				} else {
+					const fullUri = vscode.Uri.joinPath(vscode.workspace.workspaceFolders![0].uri, url);
+					const text = fs.readFileSync(fullUri.fsPath).toString('base64');
+					const type = mime.lookup(fullUri.fsPath) || "image/png";
+					return { type: 'image_url', image_url: { url: `data:${type};base64,${text}` } };
+				}
+			});
+			this.chatMessages.push({ role: role, content: content });
+		}
```

Comment:
I decided to implement the image attachment feature in `addChatMessage(...)`, to avoid making broad changes to the interface.

Reply:
I think it's a good decision.